### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by masterleinad**
This pull request removes all the code guarded by `KOKKOS_ENABLE_DEPRECATED_CODE` to ease the burden of supporting multiple configurations.
The `Makefile/CMake` options are also removed but I am open to keeping them for future use.

I didn't indent the first commit on purpose to make reviewing easier.

### CODE REVIEW

File: cmake/kokkos_enable_options.cmake [hunk 1]
```diff
@@ -48,7 +48,6 @@ KOKKOS_ENABLE_OPTION(COMPILER_WARNINGS    OFF "Whether to print all compiler war
 KOKKOS_ENABLE_OPTION(PROFILING            ON  "Whether to create bindings for profiling tools")
 KOKKOS_ENABLE_OPTION(PROFILING_LOAD_PRINT OFF "Whether to print information about which profiling tools got loaded")
 KOKKOS_ENABLE_OPTION(AGGRESSIVE_VECTORIZATION OFF "Whether to aggressively vectorize loops")
-KOKKOS_ENABLE_OPTION(DEPRECATED_CODE          OFF "Whether to enable deprecated code")
```
Comment (ndellingwood) [inline cmake/kokkos_enable_options.cmake:51]: @masterleinad is it reasonable to keep the DEPRECATED_CODE macro available for future code deprecation (i.e. in anticipation of Kokkos 4.0)?

Comment (masterleinad) [reply cmake/kokkos_enable_options.cmake:51]: I am open to keep it, or remove and reintroduce again when necessary.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Any opinions on this? Is it too early to think about it? In that case, I am happy to close.

**Comment by dalg24** [comment]
> Any opinions on this? Is it too early to think about it? In that case, I am happy to close.

This needs more discussion.  It is on the agenda for the upcoming developer meeting.

**Comment by masterleinad** [comment]
Rebased.

**Review by crtrott — APPROVED** [review]
Went through every line.

**Pull request merged**